scripts/security/scan.sh

```
@@ -0,0 +1,278 @@
+usage() {
+    grep "^# Usage:" "$0" | sed 's/^# //'
+    exit 2
+}
+
+log_info()  { echo -e "${BLUE}[INFO]${NC}  $*"; }
+log_pass()  { echo -e "${GREEN}[PASS]${NC} $*"; }
+log_warn()  { echo -e "${YELLOW}[WARN]${NC}  $*"; }
+log_error() { echo -e "${RED}[ERROR]${NC} $*"; }
+
+check_pass() { ((PASS_COUNT++)); echo -e "  ${GREEN}PASS${NC} $1"; }
+check_fail() { ((FAIL_COUNT++)); echo -e "  ${RED}FAIL${NC} $1"; }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)